src/SqlVirtualMachine/SqlVirtualMachine.Autorest/custom/Assert-AzSqlVMADAuth.ps1

```diff
@@ -0,0 +1,520 @@
+    .SYNOPSIS
+    Validate the provided MSI has required permissions or not
+	
+	.Description
+	Validate the provided MSI has required permissions or not
+	
+	.PARAMETER PrincipalId
+    Msi Principal Id
+#>
+function Assert-MsiWithEnoughPermission {    
+	param(
+        [Parameter(Mandatory = $true)]
+        [string] $PrincipalId
+    )
+	[Microsoft.Azure.PowerShell.Cmdlets.SqlVirtualMachine.DoNotExportAttribute]
+    
+		# Install and connect to Microsoft Graph
+		Install-Module Microsoft.Graph.Authentication -Scope CurrentUser -AllowClobber -Force
+		$cred = Get-Credential
+		$tenantId = (Get-AzContext).Subscription.TenantId
+		Connect-MgGraphViaCred -credential $cred -tenant $tenantId
+
+		# Install and import Microsoft.Graph.Applications module
+		Install-Module -Name Microsoft.Graph.Applications -Scope CurrentUser -AllowClobber -Force
+		Import-Module Microsoft.Graph.Applications
+
+		# Get directory roles assigned to the MSI
+		$directoryRoles = Get-DirectoryRoleList -PrincipalId $PrincipalId
+
+		# Check if the MSI has the "Directory Readers" role
+		if ($directoryRoles.DisplayName -contains "Directory Readers") {
+			return
+		}
+
+		# Retrieve app role IDs for required roles
+		$appRoleIdMap = Find-RoleId
+
+		# Retrieve all assigned app role IDs for the MSI
+		$allAssignedRoleIds = (Get-MgServicePrincipalAppRoleAssignment -ServicePrincipalId $PrincipalId).AppRoleId
+
+		# Find missing roles
+		$missingRoles = @("User.Read.All", "Application.Read.All", "GroupMember.Read.All") | Where-Object { $appRoleIdMap[$_] -notin $allAssignedRoleIds }
+
+		if ($missingRoles.Count -gt 0) {
+			$azError = "The managed identity is lacking the following roles for Azure AD authentication: $($missingRoles -join ', ')."
+			$azError += "`n Recommendation: Grant the managed identity EITHER the Directory.Readers role OR the three App roles 'User.Read.All', 'Application.Read.All', 'GroupMember.Read.All'"
+			throw $azError
+		}		
+}
+
+  
+function Connect-MgGraphViaCred {
+      <#
+      .SYNOPSIS
+      Function for connecting to the Microsoft Graph using given credentials.
+      This option is unavailable with official Connect-MgGraph command.
+
+      .DESCRIPTION
+      Function for connecting to the Microsoft Graph using given credentials.
+      This option is unavailable with official Connect-MgGraph command.
+
+      .PARAMETER credential
+      Credential object.
+
+      .PARAMETER tenant
+      (optional) Azure tenant name or id.
+
+      .EXAMPLE
+      $cred = Get-Credential
+      Connect-MgGraphViaCred -credential $cred
+      #>      
+      param(
+          [Parameter(Mandatory = $true)]
+          [System.Management.Automation.PSCredential] $credential,
+
+          [string] $tenant = $_tenantDomain
+      )
+	  [Microsoft.Azure.PowerShell.Cmdlets.SqlVirtualMachine.DoNotExportAttribute]
+
+		try {
+			# Connect to Azure using credentials
+			$null = Connect-AzAccount -Credential $credential -TenantId $tenant -Force -ErrorAction Stop
```

Comment:
Hi @YanaXu - 
Since I have a dependency on interactive login for credentials, it's not allowing me to record tests.
Is there any workaround for this issue?

Reply:
> Hi @YanaXu - Since I have a dependency on interactive login for credentials, it's not allowing me to record tests. Is there any workaround for this issue?

I have figured out a work around and added tests..